v0/destinations/am/transform.js

```diff
@@ -208,14 +195,68 @@ function responseBuilderSimple(
     default:
       set(rawPayload, "event_properties", message.properties);
       rawPayload.event_type = evType;
+      if (
```

Comment:
The code would be more concise, modular and readable, if we move the below code in some other function.